platform/ios/src/MGLMapView.mm

```diff
@@ -2341,6 +2346,88 @@ - (void)setShowsScale:(BOOL)showsScale
     }
 }
 
+- (void)setShowsBuildings:(BOOL)showsBuildings
+{
+    if  (_showsBuildings == showsBuildings) return;
+    
+    NSString *layerIdentifier = @"extrudedBuildings";
+    if (showsBuildings) {
+        MGLSource *source = [self.style sourceWithIdentifier:@"composite"];
+        if (source) {
+            
+            MGLFillExtrusionStyleLayer* layer = [[MGLFillExtrusionStyleLayer alloc] initWithIdentifier:layerIdentifier source:source];
+            layer.sourceLayerIdentifier = @"building";
+            layer.predicate = [NSPredicate predicateWithFormat:@"extrude == 'true' AND CAST(height, 'NSNumber') > 0"];
+            layer.fillExtrusionBase = [NSExpression expressionForKeyPath:@"min_height"];
+            layer.fillExtrusionHeight = [NSExpression expressionForKeyPath:@"height"];
+            
+            // Set the fill color to that of the existing building footprint layer, if it exists.
+            MGLFillStyleLayer* buildingLayer = (MGLFillStyleLayer*)[self.style layerWithIdentifier:@"building"];
+            if (buildingLayer) {
+                if (buildingLayer.fillColor) {
+                    layer.fillExtrusionColor = buildingLayer.fillColor;
+                } else {
+                    layer.fillExtrusionColor = [NSExpression expressionForConstantValue:[UIColor whiteColor]];
+                }
+                
+                layer.fillExtrusionOpacity = [NSExpression expressionForConstantValue:@0.75];
+            }
+            
+            MGLStyleLayer* labelLayer = [self.style layerWithIdentifier:@"waterway-label"];
+            if (labelLayer) {
+                [self.style insertLayer:layer belowLayer:labelLayer];
+            } else {
+                [self.style addLayer:layer];
+            }
+        }
+    } else {
+        MGLStyleLayer *layer = [self.style layerWithIdentifier:layerIdentifier];
+        if (layer) [self.style removeLayer:layer];
```

Comment:
This is a bit destructive. We should instead hide the layer so it can be easily shown again later if the property gets toggled again later.